build/ci/release/Jenkinsfile

```diff
@@ -12,53 +12,41 @@ pipeline {
         VAULT_ADDR = credentials('vault-addr')
         VAULT_ROLE_ID = credentials('vault-role-id')
         VAULT_SECRET_ID = credentials('vault-secret-id')
-        REPOSITORY = "eck"
-        IMG_NAME = "eck-operator"
-        REGISTRY = "docker.elastic.co"
-        OPERATOR_IMAGE = "${REGISTRY}/${REPOSITORY}/${IMG_NAME}:${TAG_NAME}"
+
+        TAG_NAME = sh(script: 'echo -n $TAG_NAME', returnStdout: true)
     }
 
     stages {
-        stage('Doing release') {
-            when {
-                expression {
-                    releaseImageNotExist()
-                }
-            }
+        stage('Nightly or release build') {
             stages {
-                stage('Build and push release image') {
+                stage('Build and push Docker image') {
                     steps {
                         sh """
-                            cat >.env <<EOF
-VERSION = ${TAG_NAME}
-REGISTRY = ${REGISTRY}
-REPOSITORY = ${REPOSITORY}
-IMG_NAME = ${IMG_NAME}
-OPERATOR_IMAGE = ${REGISTRY}/${REPOSITORY}/${IMG_NAME}:${TAG_NAME}
-SNAPSHOT = false
-GO_TAGS = release
-export LICENSE_PUBKEY = /go/src/github.com/elastic/cloud-on-k8s/build/ci/license.key
-ELASTIC_DOCKER_LOGIN = eckadmin
-EOF
-                            make -C build/ci get-docker-creds get-elastic-public-key TARGET=ci-release ci
+                        build/ci/setenv
+                        make -C build/ci get-docker-creds get-elastic-public-key TARGET=ci-release ci
                         """
                     }
                 }
                 stage('Upload yaml to S3') {
+                    when {
+                        buildingTag()
+                    }
                     steps {
                         sh 'make -C build/ci yaml-upload'
                     }
                 }
-                stage('Send message to Slack') {
+                stage('Notify successful release build') {
+                    when {
+                        buildingTag()
```

Comment:
Is this built in? I don't see where it is defined.

Reply:
Yes, it's a built-in condition of the declarative pipeline: https://jenkins.io/doc/book/pipeline/syntax/#built-in-conditions